saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/Saml2PostAuthenticationRequest.java

```diff
@@ -50,7 +51,8 @@ public Saml2MessageBinding getBinding() {
 	 */
 	public static Builder withRelyingPartyRegistration(RelyingPartyRegistration registration) {
 		String location = registration.getAssertingPartyDetails().getSingleSignOnServiceLocation();
-		return new Builder().authenticationRequestUri(location);
+		return new Builder().authenticationRequestUri(location)
```

Comment:
Instead of calling a mutator method, I think that the `Builder` constructor should take a registration id as a value and that value should be immutable.